GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -364,12 +365,30 @@ protected virtual bool TryFetchReleases(out List<Release> releases, out string e
             return false;
         }
 
-        protected virtual void RunInstaller(string path, string args, out int exitCode, out string error)
+        protected virtual void RunInstaller(string path, string args, string certCN, out int exitCode, out string error)
         {
-            ProcessResult processResult = ProcessHelper.Run(path, args);
+            Stream stream = this.fileSystem.OpenFileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, false);
 
-            exitCode = processResult.ExitCode;
-            error = processResult.Errors;
+            if (stream == null)

```

Comment:
This stream check seems like something you should do as part of the `VerifyCodeSignature`, as you are not using it any more after this check.

Also, stick it in a `using()` statement so you are sure to close it even on exceptions.